### PR TITLE
inline ure form on upload page.

### PR DESCRIPTION
**Pull request by akash1810**
inline-ure

### CODE REVIEW

File: kahuna/public/stylesheets/main.css [hunk 1]
```diff
@@ -1181,31 +1181,29 @@ FIXME: what to do with touch devices
...
-.result-editor__labeller-container gr-add-label {
+.result-editor__field-container gr-add-label {
     font-size: 1.3rem;
     padding-right: 10px;
 }
 
-.result-editor__label {
+.result-editor__field-label {
     width: 100px;
+    flex-shrink: 0;
```
Comment (akash1810) [inline kahuna/public/stylesheets/main.css]: this took far too long to work out! :crying_cat_face:

**Comment by akash1810** [comment]
not complete - css woes...

screen shot 2015-08-20 at 17 43 18

**Comment by theefer** [comment]
Nice start! It feels like it would be nice to show the current value even when the panel is collapsed, otherwise you don't know whether you need to expand it or not without expanding it (we also talked of having it expanded by default if no usage rights present -- that'd be complementary).

Should we try to reuse the layout (rights above metadata) and editing activation (the pen) as on other metadata panels?

image

**Comment by akash1810** [comment]
> It feels like it would be nice to show the current value even when the panel is collapsed

and 

> having it expanded by default if no usage rights present

Agreed! Feels like a separate PR to me.

Would have done the latter as part of this PR, but it took stupidly long to work out the why the css wasn't playing nice :neutral_face: 

> Should we try to reuse the layout (rights above metadata)

I did that, but it looked strange... maybe because I'm used to the other fields being fist on the Uploads.

> and editing activation (the pen) as on other metadata panels

:+1:

**Comment by theefer** [comment]
:+1: Let's chat tomorrow about the next steps ^

**Comment by akash1810** [comment]
Updated now to:

- Auto expand the form when there are no usage rights on an image
- Follow the same design pattern of hover to reveal pencil -> click pencil to enter edit mode

inline-ure

File: kahuna/public/stylesheets/main.css [hunk 2]
```diff
@@ -282,7 +282,7 @@ a:focus {
     width: 100%;
 }
 
-.right {
+.flex-right {
```
Comment (akash1810) [inline kahuna/public/stylesheets/main.css]: meep, this shouldn't really be part of this PR. `flex-right` felt like a more appropriate name for this class.

**Comment by theefer** [comment]
Looks like "Usage rights category" doesn't fit. Should we just use Usage rights here (and maybe elsewhere)?

**Comment by theefer** [comment]
Sorry to nitpick but the pen at the far right is very far from any content, could easily be overlooked. What do you think?

**Comment by akash1810** [comment]
> Should we just use Usage rights here

:+1: looks nicer (will fix alignment issues)

screen shot 2015-08-21 at 12 06 15

**Comment by akash1810** [comment]
> could easily be overlooked

By design. My thinking was, if there aren't any usage rights, the form will already be expanded.

If there are initial usage rights, then its what the system thinks they should be - having the edit far away should mean less accidental changes (unsure if this happens IRL). There might also be an argument in that if we keep manually overriding the system usage rights, the system's formula might actually be wrong.

**Comment by akash1810** [comment]
> but the pen at the far right is very far from any content

@alastairjardine had suggested a nice change to the edit pattern on the metadata panels - on hover, the entire row will highlight to better distinguish what field the pencil belongs to. Might see how that looks.

**Comment by theefer** [comment]
Not convinced about making it artificially confusing to edit. If the user wants to take the action, it should be easy to figure out how?

Is the usage rights category text smaller on purpose? Can it be normal text style?

**Comment by akash1810** [comment]
Can't quite get row highlight to work properly across the site - additional PR.

Changes @theefer suggested:

screen shot 2015-08-21 at 13 15 15

The above is the hover state.

File: kahuna/public/js/edits/image-editor.js [hunk 3]
```diff
@@ -75,6 +77,16 @@ imageEditor.controller('ImageEditorCtrl',
 
 
     ctrl.usageRights = imageService(ctrl.image).usageRights;
+
+    function updateUsageRightsCategory() {
+        let category = ctrl.categories.find(cat => cat.value === ctrl.usageRights.data.category);
+        ctrl.usageRightsCategory = category && category.name;
+        ctrl.showUsageRights = ctrl.usageRightsCategory === undefined;
```
Comment (theefer) [inline kahuna/public/js/edits/image-editor.js]: Is that going to expand it every time the user saves `None` as a usage right?

**Comment by theefer** [comment]
:+1: Looks like another step in the right direction. Can you ping us when it's on TEST for one last look?

File: kahuna/public/js/edits/image-editor.js (hunk 3 above)
Comment (akash1810) [inline kahuna/public/js/edits/image-editor.js]: In theory, yes. However, `None` isn't an option...

screen shot 2015-08-21 at 14 22 39

**Comment by akash1810** [comment]
on TEST now.

**Comment by theefer** [comment]
So once you've set a category you can't unset it?

Seems like a usability issue to me. Can we fix in another branch, or at least discuss (Slack?)?

**Comment by theefer** [comment]
Tried this on TEST, looks great!
